### ISSUE TITLE
Make 'Material You - Current' widget look alike Google's one

### ISSUE THREAD

**Issue by nit511** [breezy-weather/breezy-weather#316]
### Describe your idea

I know i am nitpicking, but i use this app on daily basis and wants to be it perfect.
I request to make the 'Material You - Currnet' Widget looks more like the Google's weather widget because it looks more beautiful on home screen.

![20230807_150702](https://github.com/breezy-weather/breezy-weather/assets/103308339/8a5ccddc-a4d0-4a28-a9b7-0c0fe9f22b8c)
*Left one is Breezy's and Right one is Google's

Differences:-
1. Size (height, width) of widget is less and should be increase. (Red)
2. Temperature degree number is small and should be aligned in horizontally middle of widget. (Green)
3. Weather logo is small and need some position alignment. (Blue)
4. Temperature degree number's material you color is different than Google's widget.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] My idea is not related to an existing feature described in the [Homepage](https://github.com/breezy-weather/breezy-weather/blob/main/docs/HOMEPAGE.md) document.
- [X] I have updated the app to **[latest version](https://github.com/breezy-weather/breezy-weather/releases/latest)**.

**Comment by papjul** [comment]
We are currently redesigning widgets and we are unsure if this one will be kept as it makes a poor usage of available space. Using Material You color for text will probably be on the list though.

**Comment by M00NJ** [comment]
I somehow missed this issue and opened a duplicate with similar suggestions. I will paste it again here as there are some additional suggestions and mock-ups in it:

### Describe your idea:

I compared the Material You widget to the one by Google and noticed some things that could be improved.

Breezy (Left) / Google (Right):

![Image1](https://github.com/breezy-weather/breezy-weather/assets/82398591/ddd58ad3-cb91-4466-8464-eeca38de0e6d)

1. Change font color to `primary`  
    (it's `onsecondary` now)
2. Align text to `center` 
    (it's aligned to the `right` now)
3. Use `regular` font
    (it's `bold` now)
4. Change font size based on string length
5. Make widget resizable

Making sure the temperature text is aligned properly might require some logic to always place it inside the green circle and possibly reduce the font size, if the string is longer because Fahrenheit or Kelvin is being used. I created some mock-ups to show what I mean:

![Image2](https://github.com/breezy-weather/breezy-weather/assets/82398591/c49c79b3-5ed7-4a1c-8638-d885098ec6f7)

Mock-up without the circles:

![Image3](https://github.com/breezy-weather/breezy-weather/assets/82398591/4bd5b022-3012-49e9-81a4-30ac79ce335c)

The google widget is also resizable:

![Image4](https://github.com/breezy-weather/breezy-weather/assets/82398591/d3d43943-bd3d-4a7c-8c96-a5b10583db59)

**Comment by M00NJ** [comment]
> We are currently redesigning widgets and we are unsure if this one will be kept as it makes a poor usage of available space.

I agree, the widget is unnecessarily big. But the style and simplicity is quite nice imo. So making it resizable should fix the concern. Then the space inefficiency would be a user choice.

**Comment by papjul** [comment]
Some improvements to the proportions are coming in next version

**Comment by jfk93260** [comment]
Hello,
I know you tried to fix issues with proportions for the Material You widget but... was it really necessary to make it THAT big ? The widget now takes twice the space it used to and doesnt' seem to fit my grid or be resizable.
That's the old one:
![Screenshot_20240331-013946-picsay](https://github.com/breezy-weather/breezy-weather/assets/30798957/ed6b38dc-9a9c-4d6e-bc13-8837734f33a9)
And the new one:
![Screenshot_20240331-013347-picsay](https://github.com/breezy-weather/breezy-weather/assets/30798957/e147fe1e-e63e-498a-8ba5-0375ace85aa1)
If I wanted to display the entirety of the widget I would really need to make it twice as long both in height and width...
The widget also stopped reacting to taps and doesn't open the app anymore.

**Comment by papjul** [comment]
> I know you tried to fix issues with proportions for the Material You widget but... was it really necessary to make it THAT big ?

It's normal-sized, it's not bigger than Google's. Put the Google one by its side and you will see that it's the exact same size.

> The widget also stopped reacting to taps and doesn't open the app anymore.

I can look into it.

**Comment by jfk93260** [comment]
The problem is that, on the opposite of Google Material You widgets, yours doesn't adapt its size to the space it's been allowed.
I don't have the Google Weather widget so I can't compare but here is what Google Clock Material You widgets look like when you resize them
![Screenshot_20240331-084516-picsay](https://github.com/breezy-weather/breezy-weather/assets/30798957/11cb57de-6d68-4233-8b23-1eac01ced41c)
![Screenshot_20240331-084511-picsay](https://github.com/breezy-weather/breezy-weather/assets/30798957/684e281a-d778-458f-9eba-62c52c77b303)
![Screenshot_20240331-084539-picsay](https://github.com/breezy-weather/breezy-weather/assets/30798957/01d88e81-8870-436a-b5c5-ebaf2ddae652)
Your old one was also able to do that 
![Screenshot_20240331-013946-picsay](https://github.com/breezy-weather/breezy-weather/assets/30798957/4233e4f9-9027-4ab9-b71d-268bcdbda2dc)
![Screenshot_20240331-032305-picsay](https://github.com/breezy-weather/breezy-weather/assets/30798957/d1026436-6a60-4cb6-974a-327b1e01f586)
Your new one doesn't. If I try to resize it, it just crops to fit as you can see in the screenshot
![Screenshot_20240331-013347-picsay](https://github.com/breezy-weather/breezy-weather/assets/30798957/4c22f87c-0f5e-4e78-a58b-0bae9e79e07f)
I'm sorry but I CAN'T possibly see that new behaviour as an improvement

**Comment by papjul** [comment]
> The problem is that, on the opposite of Google Material You widgets, yours doesn't adapt its size to the space it's been allowed.

The Google Weather widget adapts its size to the space allowed but not in the way you think. See the end of [this message](https://github.com/breezy-weather/breezy-weather/issues/316#issuecomment-1773518554). That's one of the remaining works to do.

> I don't have the Google Weather widget

It's part of the Google app, just like the clock

**Comment by jfk93260** [comment]
> It's part of the Google app, just like the clock

No, it's not. At least not for everybody. My Google App couldn't be more up-to-date but still the Material You Weather widget is not part of the included widgets. As for the clock widget, it's definitely part of the Google Clock app... But that doesn't really matter anyway

> Or if it did, it's a feature of your launcher.

Yes, my bad. Square Home is indeed able to resize, constraint and apply margins on widgets but for some reason it's not able to do that on your updated version of the Material You widget. I never realised the old one was not even resizable. After some tweaking I've been able to achieve the same result... Well,  more or less ^^
Anyway, thanks for your work on this fork, I'm really enjoying Breezy Weather 👍

**Comment by papjul** [comment]
> My Google App couldn't be more up-to-date but still the Material You Weather widget is not part of the included widgets

That's curious because the widget is definitely in the manifest (and has been for years). Maybe they restricted it to some Android versions or something. I have no idea, sorry :/ Usually, I would say "ask the developer" but in that case, you might have to wait a long time ^^'

> Square Home is indeed able to resize, constraint and apply margins on widgets

Oh, good to know, I will keep that in mind if other users ask :)

> Anyway, thanks for your work on this fork, I'm really enjoying Breezy Weather 👍

:)

**Comment by papjul** [comment]
A few updates coming in v5.1.9:
- Adapt the widget to its frame on Android >= 12
- Preview of the widget when adding widget
- Improve temperature text size depending on the number of digits
- Improve weather icon size

Remains to do:
- Use correct Material You color
- Resize to pill shape
